src/lightning/pytorch/demos/transformer.py

```diff
@@ -60,20 +64,29 @@ class PositionalEncoding(nn.Module):
     def __init__(self, dim: int, dropout: float = 0.1, max_len: int = 5000) -> None:
         super().__init__()
         self.dropout = nn.Dropout(p=dropout)
+        self.dim = dim
+        self.max_len = max_len
 
-        pe = torch.zeros(max_len, dim)
-        position = torch.arange(0, max_len, dtype=torch.float).unsqueeze(1)
-        div_term = torch.exp(torch.arange(0, dim, 2).float() * (-math.log(10000.0) / dim))
-        pe[:, 0::2] = torch.sin(position * div_term)
-        pe[:, 1::2] = torch.cos(position * div_term)
-        pe = pe.unsqueeze(0).transpose(0, 1)
+        pe = self._init_pos_encoding()
         # workaround, can't use buffer, see https://github.com/pytorch/pytorch/issues/68407
         self.register_parameter("pe", nn.Parameter(pe, requires_grad=False))
 
+    def reset_parameters(self) -> None:
+        self.pe.copy_(self._init_pos_encoding())
```

Comment:
mypy has issues with this line of code 🤔

Reply:
lets add ignore :)
```suggestion
        self.pe.copy_(self._init_pos_encoding())  # type: ignore
```